### PR TITLE
fix 306

### LINKED ISSUE #306: Ability to scout playoff matches
We should have the ability to scout playoff matches (perhaps we're video scouting a different event).
We need a way to indicate "match number" for playoffs.
Do we want to know upper/lower bracket? (do we care?)

### PR DESCRIPTION
**Pull request by Jeff-Stammler**
fixes #306

If you test this, you MUST first delete the Matches.csv file in "CPR Scouting/Input" on your AVD.

2 large changes to make this work:
whole new class on MatchTypes (quals, etc).
Matches class is now an Array of Arrays, so that was most of the logic of knowing which matchlist LIST to reference when trying to populate matches.  I did this in CASE we get update match schedules based on Semi's or Finals (if not soon, then maybe down the road when we can auto-update match schedules).

Pre-Match.xml has layout and spinner things, but standard changes.

Closing #306

### CODE REVIEW

File: app/src/main/java/com/team3663/scouting_app/data/MatchTypes.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    // Class:       MatchTypeRow
+    // Description: Defines a structure/class to hold the information for each match type
+    // =============================================================================================
+    private static class MatchTypeRow {
+        private final int id;
+        private final String short_form;
+        private final String description;
+
+        // Constructor with individual data
+        public MatchTypeRow(String in_id, String in_short_form, String in_description) {
```
Comment (AdrielStammler) [inline app/src/main/java/com/team3663/scouting_app/data/MatchTypes.java:98]: Why is in_id a `String` and not a `int`? You immediately convert it to an `int` so why not just take in an int and make it simpler?

> Line 18
> `public void addMatchTypeRow(String in_id, String in_short_form, String in_description) {`


_(and then also make this in_id an `int` too)_

File: app/src/main/java/com/team3663/scouting_app/activities/AppLaunch.java [hunk 2]
```diff
@@ -122,6 +122,9 @@ protected void onCreate(Bundle in_savedInstanceState) {
         }
         else
             initStoragePermissions();
+
+        // While loading Matches, we messed with Globals.CurrentMatchType, so reset it
+        Globals.CurrentMatchType = 1;
```
Comment (AdrielStammler) [inline app/src/main/java/com/team3663/scouting_app/activities/AppLaunch.java:127]: No hardcoded numbers

**Review by AdrielStammler — APPROVED** [review]

File: app/src/main/java/com/team3663/scouting_app/data/MatchTypes.java (hunk 1 above)
Comment (Jeff-Stammler) [reply app/src/main/java/com/team3663/scouting_app/data/MatchTypes.java:98]: It's to make the loaddata fn() generic with no knowledge of what the file specific data it loaded.  It was like this before my check in.  
All conversions happen in the function calls.

Comment (AdrielStammler) [reply app/src/main/java/com/team3663/scouting_app/data/MatchTypes.java:98]: Ok

**Review by AdrielStammler — APPROVED** [review]

**Pull request merged**